src/core/libmaven/PeakGroup.cpp

```diff
@@ -912,3 +927,20 @@ void PeakGroup::setSelectedSamples(vector<mzSample*> vsamples){
         }
     }
 }
+
+void PeakGroup::setAdduct(Adduct* adduct)
+{
+    _adduct = adduct;
+    if (_adduct != nullptr
+        && _adduct->getName() != MassCalculator::MinusHAdduct->getName()
+        && _adduct->getName() != MassCalculator::PlusHAdduct->getName()) {
+        _type = GroupType::Adduct;
+    }
+}
+
+Adduct* PeakGroup::getAdduct() const
+{
+    if (isIsotope())
+        return parent->getAdduct();
```

Comment:
Wondering if we should even calculate adduct isotopes. What's the formula you're using for adduct groups?

Reply:
I don't think we intended to calculate adduct isotopes, explicitly. But to answer your question, the original compound's formula is used for enumerating labels and then the parent-ion's mass (which is itself adjusted with adduct mass) is added with respective isotope masses.